UI updates

c-a-ray opened the pull request:
This PR fixes a handful of UI things that have been bugging me. Its goal is to make the app look cleaner and more consistent. Main things it does:
- Adds consistent dark mode styling across the entire app
- Adds a shared component for status messages
- Makes it so updating a role in the Admin page dynamically updates the sidebar, without a page refresh

Some small incidentals it also takes care of:
- Aligns the buttons on the login page
- Updates the tab title to "BeaverHealth" (previously read "Laravel")
- Includes the uploaded filename in the Add Records status message
- Removes unused files (mostly unused Blade templates)

UI updates:

Login page buttons (feel like this could maybe be better, but haven't found anything I really like)
![image](https://github.com/user-attachments/assets/25f15a5b-0905-4d82-8409-768ef8c70697)

- Sidebar updates
![image](https://github.com/user-attachments/assets/89cc9225-789a-46cd-a57a-5f8c8930f0c3)

- Admin page table
![image](https://github.com/user-attachments/assets/c712369d-b176-4a64-93a8-09b6a24d5b8f)
![image](https://github.com/user-attachments/assets/ab99d75a-b538-458f-9124-770b1e14f859)

- Vulnerability Toggles table
![image](https://github.com/user-attachments/assets/a8b3cec4-4c36-4815-8b34-ef65c1c27642)
![image](https://github.com/user-attachments/assets/fc7d4ec0-755d-488a-aec5-ed14e458eb07)

- Patient Info table
![image](https://github.com/user-attachments/assets/425839e5-b14d-463f-a425-a748eb3f49f9)
![image](https://github.com/user-attachments/assets/17dc9ff7-edc7-49ab-b359-bc89e041d987)

- Patient Feedback dark mode
![image](https://github.com/user-attachments/assets/3e0b28f0-a134-4a1b-8482-16501778e525)
![image](https://github.com/user-attachments/assets/27781887-c1cc-4444-840e-80cc989ba144)

- Container for Change Password form
![image](https://github.com/user-attachments/assets/ed367b5f-2858-48e0-9941-0b6ac11013c9)
![image](https://github.com/user-attachments/assets/72789daa-33d9-4c57-b1e3-56e0b1d24dd7)

- Consistent status message popups across the app
![image](https://github.com/user-attachments/assets/f3162b05-adf7-45bf-b56b-0270adc22811)
![image](https://github.com/user-attachments/assets/4e2701af-799c-4af2-b9ca-d381fa36cf27)
![image](https://github.com/user-attachments/assets/ee603f9e-b475-4302-a4e2-3ebe2971a48d)
![image](https://github.com/user-attachments/assets/a89e7208-80eb-494e-b3dd-23ac53d3fe1a)

c-a-ray commented on resources/js/vulnerability_toggles.js:
```diff
@@ -1,61 +1,70 @@
-    } else {
-        notification.querySelector('div').className =
-            'px-4 py-2 rounded-lg shadow-lg text-center min-w-[200px] bg-red-100 text-red-800 dark:bg-red-800 dark:text-red-100';
+document.addEventListener('DOMContentLoaded', function() {
```
Initially edited this file to move the notification to the top right, improve dark-mode support, and fix the error styles not being applied, but then figured I might as well do a proper refactor by:
- Centralizing notification styles in a config object
- Adding proper timeout/cleanup management 
- Adding smooth fade animations
- Improving error handling and state recovery
- Converting to async/await pattern

c-a-ray commented on resources/views/profile/change-password.blade.php:
```diff
@@ -5,60 +5,65 @@
-                    {{ __('Ensure your password is at least 8 characters long.') }}
-                </p>
-            </header>
+    <div class="py-12">
```
Adds a container for the form to make it consistent with the rest of the app.

BrynnMartina replied:
Another potential area we could universalize. You could create a container component 

![image](https://github.com/user-attachments/assets/623f38d5-b2ec-4981-90bd-4c99b177bf4f)

which would allow you to replace `<div class="max-w-7xl mx-auto sm:px-6 lg:px-8">` with `<x-container>`

I don't see this as being as big of a priority as other elements, but you get the point.

c-a-ray replied:
I looked into creating a container component but it turned out to actually be kind of difficult. There are a lot of styling variations for containers between pages, so a single reusable component would need a lot of parameters or overrides. Just seems like it's more trouble than it's worth.

Not opposed to the idea if you want to give it a shot though.

BrynnMartina reviewed (commented):
Overall, these modifications look great and I don't see anything holding it back from being merged. The modifications make the website look much more cohesive. UI has been on the back burner in the grand scheme of things, and I think there's an opportunity to talk more about this at our weekly meeting. For the future there may be some perks towards making some of these UI features more universally accessible. And this job does not necessarily have to fall to you.

For example, success and error messages are not currently consistent between all pages. We could utilize resources/views/components more, similar to the patient-dropdown.blade.php file made for the dropdown listing patient names and SSN.

We could make a new file, like notification.blade.php:

![Screenshot from 2025-02-24 01-52-13](https://github.com/user-attachments/assets/e561e08a-064a-4f6e-b7a1-432939e902da)

Add a component in add.blade.php:

![image](https://github.com/user-attachments/assets/b4864ae4-e231-47a1-83ba-e442ad2468bd)

And have the controllers handle the rest (which some pages already do currently)
`return back()->with('success', 'File uploaded successfully!');`


Unrelated:
On the login page the left side of the "Log In" button does not align with the "Username" and "Password" text boxes. To align it, you can change `ms-3` to `ml-0`. Just a personal preference of mine.

Login.blade.php:
![image](https://github.com/user-attachments/assets/ccbc1c05-aef2-45d3-b0ce-46ca7604b93f)

c-a-ray commented:
> For example, success and error messages are not currently consistent between all pages. We could utilize resources/views/components more, similar to the patient-dropdown.blade.php file made for the dropdown listing patient names and SSN.
> 
> We could make a new file, like notification.blade.php:

That's so funny -- I didn't realize you had reviewed my PR, but I made exactly this change. Added a `status-message.blade.php` Blade template and a `StatusMessage` component class, and updated all of the pages to use it.

c-a-ray commented:
> Unrelated: On the login page the left side of the "Log In" button does not align with the "Username" and "Password" text boxes. To align it, you can change `ms-3` to `ml-0`. Just a personal preference of mine.

I like your thinking here. You're right that the Login page buttons could look better. I was playing around with it and landed on this:
![image](https://github.com/user-attachments/assets/ded83ad3-20d8-4dae-88b5-fef51c85232a)

What do you think?

Using `ml-0` does look better than `ms-3`, for sure. But then I was trying to make it so the "CREATE AN ACCOUNT" button and the "LOG IN" button were lined up horizontally and it didn't look quite right. Centering them seemed to look the best to me, but I could definitely be convinced otherwise.

c-a-ray commented on app/Http/Requests/ProfileUpdateRequest.php:
```diff
@@ -1,28 +0,0 @@
-<?php
```
File is unused

c-a-ray replied:
Not used

c-a-ray commented on app/Http/Controllers/PasswordController.php:
```diff
@@ -8,7 +8,7 @@
 use Illuminate\View\View;
 
-class ProfileController extends Controller
+class PasswordController extends Controller
```
Just seems more intuitive, since all it does is change the password

c-a-ray commented on resources/js/utils.js:
```diff
@@ -0,0 +1,31 @@
+export function showStatusMessage(message, type = 'success') {
```
This `showStatusMessage` function is needed for pages that do client-side updates (AJAX). The status message component can't be used in those cases because it is rendered server-side. Currently, that's only the Vulnerability Toggles page, but I'm working on a separate PR that uses it on the Admin page. Extracted it into `utils.js` so it can be shared.

c-a-ray commented on resources/views/admin.blade.php:
```diff
@@ -4,83 +4,45 @@
         </div>
     </div>
-
-    <!-- JavaScript to handle AJAX requests -->
```
Moved to `resources/js/admin.js`. Note that any changes to those files require re-running `npm run build` in the app container.

c-a-ray commented on resources/views/records/add.blade.php:
```diff
@@ -1,7 +1,7 @@
     <x-slot name="header">
         <h2 class="font-semibold text-xl text-gray-800 dark:text-gray-200 leading-tight">
-            {{ __('Upload Patient Records') }}
+            {{ __('Add Medical Records') }}
```
Updated the page titles to be consistent with the sidebar, etc. We tend to use slightly different names for things a lot, seems better to have more consistency with what we call things.

alherrera-cs commented on app/Http/Controllers/AddRecordsController.php:
```diff
@@ -23,11 +23,14 @@ public function index()
 
     public function upload(StoreRecordRequest $request)
     {
+        $file = $request->file('medical_record');
```
good catch!
